tests/step_implementers/package/test_maven_package.py

```diff
@@ -1,339 +1,590 @@
 import os
+from pathlib import Path
 
-import pytest
 from testfixtures import TempDirectory
-import yaml
+
+import sh
+import unittest
+from unittest.mock import patch
 
 from tssc import TSSCFactory
 from tssc.step_implementers.package import Maven
 
 from test_utils import *
 
-def test_mvn_quickstart_single_jar_no_pom():
-    with TempDirectory() as temp_dir:
-        temp_dir.write('src/main/java/com/mycompany/app/App.java',b'''package com.mycompany.app;
-public class App {
-    public static void main( String[] args ) {
-        System.out.println( "Hello World!" );
-    }
-}''')
-        pom_file_path = os.path.join(temp_dir.path, 'pom.xml')
-        config = {
-            'tssc-config': {
-                'package': {
-                    'implementer': 'Maven',
+def create_mvn_side_effect(pom_file, artifact_parent_dir, artifact_names):
+    """simulates what mvn does by touching files.
+    
+    Notes
+    -----
+    
+    Supports
+    
+    - mvn clean
+    - mvn install
+    
+    """
+    target_dir_path = os.path.join(
+        os.path.dirname(os.path.abspath(pom_file)),
+        artifact_parent_dir)
+
+    
+    def mvn_side_effect(*args):
+        if 'clean' in args:
+            if os.path.exists(target_dir_path):
+                os.rmdir(target_dir_path)
+        
+        if 'install' in args:
+            os.mkdir(target_dir_path)
```

Comment:
Can we refactor to use TempDirectory() instead of os calls? I'd like to avoid os.mkdir and os.rmdir as much as possible. Python has excellent support for making temp dirs.

Reply:
@kfrankli this is already happening in a temp directory, we are creating the 'target' directory inside the temp directory that the test already created, we have to mock out what mvn does here of creating the expected 'target' dir.